### DISCUSSION TITLE
Flynt 2.0 - working with Components script.js

### DISCUSSION THREAD

**Discussion by gotxx** [flyntwp/flynt#503] [Q&A]
Hi!

Great work with new Flynt 2.0 theme. I'm working on my new project, using it.

I'd like to ask a question about Component scripts workflow.

I noticed, that after making a change in any Components/**/*.js , source is not compiled and Vite is not refreshing view.
Is it a desired behavior or am I missing something ?
 
I've managed to make a solution adding new entry  './Components/**/*.js in watchFile Array in vite.config.js
But i'm not sure is it a right way. 

Shouldn't Components script partials be observed from  './assets/main.js'  entry by default like in main.scss case ?

Thank you in advance for your answer.

**Comment by timohubois** [comment]
@gotxx in Flynt 2.0 we use a [web component](https://github.com/flyntwp/flynt/blob/master/README.md#web-components) based approach. JavaScript from `script.js` files inside the component folder is [imported dynamic](https://github.com/flyntwp/flynt/blob/596b9176bcb748a025f50505fecd7aefa52a1a7c/assets/scripts/FlyntComponent.js#L159C35-L159C50) based on the [loading strategy](https://github.com/flyntwp/flynt/blob/master/README.md#javascript-modules).

With partial scripts it should work they are imported inside the `script.js` of a component.

**Reply by gotxx** [reply]
I think I understand this approach. Callbacks fuctions are fired depending on specific strategy.
However when I'm trying to change the partial script.js e.g. in NavigationBurger Component. 
I add console.log(el) and save the file. Nothing happens. No output.
Then I try to save some other js file e.g. helpers.js. Vite is compiling and everything works as expected.
My console log appears.

**Reply by timohubois** [reply]
@gotxx tested it locally and it is working on my side. For the NavigationBurger component: did you tested with a window smaller than 1023px?

**Reply by gotxx** [reply]
@timohubois Yes, outcome is the same.  
But does it make any difference for Vite's file watching process? 
When I start development 'npm start', make some modification in any component script.js I can't see any changes.
On the other hand.. when I stop it  'ctrl+c' and run it again.. changes finally appear. 
But If I add something more in new process..  it make no effect  again .. and so on.
I have to restart every time to see any changes.

**Reply by timohubois** [reply]
@gotxx you are right it makes no difference with the file watching process and I just want to be sure.

It's really strange, because it should work without any adjustments.

Does the problem also exists with a new installation of Flynt?

**Comment by thomasnavarro** [comment]
Hi,

I've already encountered this issue while using Docker as a development environment. I had to add ` './components/**/*'` to the watchFiles. However, when using localWP, the hot module replacement (HMR) works perfectly!

**Reply by timohubois** [reply]
Thanks @thomasnavarro! @gotxx maybe `usePolling: true` can help, but I have not tested it yet:
https://vitejs.dev/config/server-options.html#server-watch

**Reply by domtra** [reply]
@gotxx are you also using docker on mac for development? if so, this seems to be a fairly common vite issue. see, for example, https://github.com/vitejs/vite/issues/4116
it might help to define a fixed post for the vite dev server and expose that with docker or additionally usePolling, as @timohubois suggested.
otherwise, in a non docker environment, we haven't had any issues with hmr before. vite should be smart enough to reload all imported js without the need of additional watchFiles for the full reload plugin.

**Reply by gotxx** [reply]
@domtra Currently I'm working on windows 10 with xampp local server. 
Although I have docker installed, but im not using it in this project. 
Thank you all for response and suggestions. appreciate it very much.
I'll try to fiddle around with different setups in free time and let you know if I have any progress.
